### PR TITLE
Updates to prepare for the 1.15.1 release

### PR DESCRIPTION
**Pull request by stmcginnis**
**Issue number:**

N/A

**Description of changes:**

This adds the 1.15.0->1.15.1 migration step to Release.toml and adds the included 1.15.1 changes to the CHANGELOG.

**Testing done:**

N/A

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
+# v1.15.1 (2023-10-9)
+
+## OS Changes
+
+* Removed explicit type for mounting filesystems to allow older ext4 snapshot volumes to be mounted in newer variants that default to xfs ([#3499])
+* Update to `apiclient` Rust dependencies ([#3491])
```
Comment (arnaldo2792) [inline CHANGELOG.md]:
```suggestion
* Update `apiclient` Rust dependencies ([#3491])
```
Or 

```suggestion
* Update tugnstenite for `apiclient` ([#3491])
```

Since we didn't actually updated all the dependencies.

File: Release.toml [hunk 2]
```diff
@@ -234,7 +234,8 @@ version = "1.16.0"
     "migrate_v1.15.0_log4j-hotpatch-enabled-metadata.lz4",
     "migrate_v1.15.0_deprecate-log4j-hotpatch-enabled.lz4",
 ]
-"(1.15.0, 1.16.0)" = [
+"(1.15.0, 1.15.1)" = []
```
Comment (arnaldo2792) [inline Release.toml:237]: Are you planning to do another PR to update the version in this file, at the very top of this file?

Comment (stmcginnis) [reply Release.toml:237]: Yes, that needs to be done only on the 1.15.x branch.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,22 @@
+# v1.15.1 (2023-10-9)
+
+## OS Changes
+
+* Removed explicit type for mounting filesystems to allow older ext4 snapshot volumes to be mounted in newer variants that default to xfs ([#3499])
```
Comment (bcressey) [inline CHANGELOG.md]:
```suggestion
* Allow older ext4 snapshot volumes to be mounted in newer variants that default to xfs ([#3499])
```

**Review by bcressey — APPROVED** [review]
LGTM

**Review by arnaldo2792 — APPROVED** [review]

**Review by jpculp — APPROVED** [review]

**Review by yeazelm — APPROVED** [review]
